pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -42,6 +45,29 @@ type MongoDBSpec struct {
 	// be set for the deployment
 	// +optional
 	FeatureCompatibilityVersion string `json:"featureCompatibilityVersion,omitempty"`
+
+	// TLS configuration for both client-server and server-server communication
+	// +optional
+	TLS TLS `json:"tls"`
```

Comment:
ah, meant to say about this. In enterprise, the current structure is `spec.security.tls` as we have other `spec.security.authentication` fields as well. Do we want to do the same for community? (eg. SCRAM is coming soon)

Reply:
@rodrigovalin what do you think?

Reply:
I have a preference for `spec.tls` but not strong enough; however I think it is a good idea to maintain consistency with enterprise. Let's make this `spec.security.tls` instead.

Reply:
I have no preference, I'll change it to `spec.security.tls`!